### PR TITLE
[inductor] Implement magic methods on IR values

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #101076

This wraps `ops` into an `OpsWrapper` object which wraps any returned
IR values into an `OpsValue` instance. This allows magic methods to
be implemented and means lowerings can write mathematical expressions much more
fluently. So instead of
```python
ops.add(ops.mul(ops.mul(ops.sub(ops.mul(_Ap2, x), _Ap3), x), x), _1)
```
we can write
```python
(_Ap2 * x - _Ap3) * x * x + _1
```

And it will translate to the equivalent `ops` calls.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

### CODE REVIEW

File: torch/_inductor/virtualized.py [hunk 1]
```diff
@@ -153,21 +154,87 @@ def __getattr__(self, item):
...
+_ops = Virtualized("ops", MockHandler)
 _graph = Virtualized("graph", NullHandler)
 _fake_mode = Virtualized("fake_mode", NullHandler)
 _kernel = Virtualized("kernel", NullHandler)
 _debug = Virtualized("debug", NullHandler)
 _interpreter = Virtualized("interpreter", NullHandler)
 
 
+class OpsValue:
+    """The turn type of most ops calls. This exists so we can do operator
```
Comment (lezcano) [inline torch/_inductor/virtualized.py:166]: nit return

**Review by lezcano — APPROVED** [review]
Cool. We'll also be able to use this class in the future to track things that can be known at lowering time.

**Review by ngimel — APPROVED** [review]
Nice! 
PR description should leave somewhere in the comments (because comments are forever, and PR description has to be searched).

**Pull request closed without merging**